inc/sitemaps/class-sitemaps-cache.php

```diff
@@ -41,6 +47,190 @@ public function is_enabled() {
+	 * @param string $postfix The part after the type in the cache key. Only the length is used.
+	 *
+	 * @return string The type with a safe length to use
+	 *
+	 * @throws OutOfRangeException When there is less than 15 characters of space for a key that is originally longer.
+	 */
+	public static function get_safe_type( $type, $prefix = '', $postfix = '' ) {
+		// Length of key should not be over 53.
+		$max_length = 53;
+		$max_length -= strlen( 'timeout_' );
+		$max_length -= strlen( $prefix );
+		$max_length -= strlen( $postfix );
+
+		if ( strlen( $type ) > $max_length ) {
+
+			if ( $max_length < 15 ) {
+				/**
+				 * If this happens the most likely cause is a 'page' that is too big.
+				 * "Normally" the max_length is 24 + strlen( page ), which is unlikely to go above 10 in the first place.
+				 *
+				 * So this would not happen unintentionally..
+				 * Either by trying to cause a high server load, finding backdoors or misconfiguration.
+				 */
+				throw new OutOfRangeException(
+					__(
+						'Trying to build a safe sitemap cache key, but the postfix and prefix combination leaves too little room to do this. You are probably requesting a page that is way out of the expected range.',
+						'wordpress-seo'
+					)
+				);
+			}
+
+			$half = ( $max_length / 2 );
+
+			$first_part = substr( $type, 0, ( ceil( $half ) - 1 ) );
+			$last_part  = substr( $type, ( 1 - floor( $half ) ) );
+
+			$type = $first_part . '..' . $last_part;
+		}
+
+		return $type;
+	}
+
+	/**
+	 * Get the cache validator for the specified type
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return string Validator to be used to generate the cache key.
+	 */
+	public static function get_validator_key( $type = null ) {
+		if ( is_null( $type ) ) {
+			return 'wpseo_sitemap_cache_validator_global';
+		}
+
+		return sprintf( 'wpseo_sitemap_%s_cache_validator', $type );
+	}
+
+	/**
+	 * Get the current cache validator
+	 *
+	 * Without the type the global validator is returned.
+	 *  This can invalidate -all- keys in cache at once
+	 *
+	 * With the type parameter the validator for that specific
+	 *  type can be invalidated
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return null|string The validator for the supplied type.
+	 */
+	private static function get_validator( $type = null ) {
+		$key = self::get_validator_key( $type );
+
+		$current = get_option( $key, null );
+		if ( ! is_null( $current ) ) {
+			return $current;
+		}
+
+		if ( self::new_validator( $type ) ) {
+			return self::get_validator( $type );
+		}
+
+		return null;
+	}
+
+	/**
+	 * Refresh the cache validator value
+	 *
+	 * @param string|null $type Provide a type for a specific type validator, null for global validator.
+	 *
+	 * @return bool True if validator key has been saved as option.
+	 */
+	public static function new_validator( $type = null ) {
```

Comment:
Command functions should start with an infinitive, this make it much more clear what the function.

`create_new_validator` or `create_validator` are appropriate here.